src/Features/Core/Portable/TodoComments/AbstractTodoCommentsIncrementalAnalyzer.cs

```diff
@@ -60,26 +60,34 @@ private ImmutableArray<TodoCommentDescriptor> GetTodoCommentDescriptors(Immutabl
             }
         }
 
+        private static ITodoCommentDataService? GetTodoService(Document document)
+        {
+            // Legacy compat until TypeScript moves to EA pattern.
+            var todoService = document.GetLanguageService<ITodoCommentService>();
+            if (todoService != null)
+                return new TodoCommentServiceWrapper(todoService);
```

Comment:
TS, this ensures we still pick up your legacy impl until you move to the new impl.

Reply:
@davidwengier hopefully this will also work for VSMac? Is this landing for 17.4 or 17.5?

Reply:
When the legacy TS bits are removed we'll need a dual insertion, but looks like everything else is in the right layers of Roslyn that this should all work in VS Mac without issue.